hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotRenameRequest.java

```diff
@@ -0,0 +1,231 @@
+    SnapshotInfo toSnapshotInfo = null;
+
+    try {
+      // Acquire bucket lock
+      mergeOmLockDetails(
+          omMetadataManager.getLock().acquireWriteLock(BUCKET_LOCK,
+                                                       volumeName, bucketName));
+      acquiredBucketLock = getOmLockDetails().isLockAcquired();
+
+      mergeOmLockDetails(omMetadataManager.getLock().acquireWriteLock(SNAPSHOT_LOCK,
+                                                       volumeName, bucketName, oldSnapshotName));
+      acquiredSnapshotOldLock = getOmLockDetails().isLockAcquired();
+
+      mergeOmLockDetails(omMetadataManager.getLock().acquireWriteLock(SNAPSHOT_LOCK,
+                                                       volumeName, bucketName, newSnapshotName));
+      acquiredSnapshotNewLock = getOmLockDetails().isLockAcquired();
+
+      // Retrieve SnapshotInfo from the table
+      String toTableKey = SnapshotInfo.getTableKey(volumeName, bucketName,
+                                                    newSnapshotName);
+      toSnapshotInfo =
+          omMetadataManager.getSnapshotInfoTable().get(toTableKey);
+
+      if (toSnapshotInfo != null) {
+        // Snapshot does not exist
+        throw new OMException("Snapshot with name " + newSnapshotName + "already exist",
+            FILE_ALREADY_EXISTS);
+      }
+
+      // Retrieve SnapshotInfo from the table
+      String fromTableKey = SnapshotInfo.getTableKey(volumeName, bucketName,
+                                                 oldSnapshotName);
+      fromSnapshotInfo =
+          omMetadataManager.getSnapshotInfoTable().get(fromTableKey);
+
+      if (fromSnapshotInfo == null) {
+        // Snapshot does not exist
+        throw new OMException("Snapshot with name " + oldSnapshotName + "does not exist",
+            FILE_NOT_FOUND);
+      }
+
+      switch (fromSnapshotInfo.getSnapshotStatus()) {
+      case SNAPSHOT_DELETED:
+        throw new OMException("Snapshot is already deleted. "
+            + "Pending reclamation.", FILE_NOT_FOUND);
+      case SNAPSHOT_ACTIVE:
+        break;
+      default:
+          // Unknown snapshot non-active state
+        throw new OMException("Snapshot exists but no longer in active state",
+            FILE_NOT_FOUND);
+      }
+
+      fromSnapshotInfo.setName(newSnapshotName);
+
+      omMetadataManager.getSnapshotInfoTable().addCacheEntry(
+          new CacheKey<>(fromTableKey),
+          CacheValue.get(termIndex.getIndex()));
+
+      omMetadataManager.getSnapshotInfoTable().addCacheEntry(
+          new CacheKey<>(toTableKey),
+          CacheValue.get(termIndex.getIndex(), fromSnapshotInfo));
+
+      omResponse.setRenameSnapshotResponse(
+          OzoneManagerProtocolProtos.RenameSnapshotResponse.newBuilder()
+              .setSnapshotInfo(fromSnapshotInfo.getProtobuf()));
+      omClientResponse = new OMSnapshotRenameResponse(
+          omResponse.build(), fromTableKey, toTableKey, fromSnapshotInfo);
```

Comment:
We need to update [SnapshotChainManager#snapshotIdToTableKey](https://github.com/apache/ozone/blob/ca14ebae27f8f1ad4ea7cc57541aabc6151c08fa/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SnapshotChainManager.java#L56) after successfully renaming the snapshot.